tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs

```diff
@@ -263,4 +262,281 @@ let c = Color.Red
         """
         |> ignoreWarnings
         |> compile
-        |> shouldSucceed
\ No newline at end of file
+        |> shouldSucceed
+
+    [<Fact>]
+    let ``Obsolete attribute is taken into account when used on an type and use extension method`` () =
+        Fsx """
+
+open System
+open System.Runtime.CompilerServices
+
+[<Obsolete("Use B instead", true)>]
+type Button = { Text : string }
+
+[<Extension>]
+type ButtonExtensions =
+
+    [<Extension>]
+    static member inline text(this: Button, text: string) =
+        { this with Text = text }
+        """
+        |> ignoreWarnings
+        |> compile
+        |> shouldFail
+        |> withErrorCode 101
+        |> withErrorMessage "This construct is deprecated. Use B instead"
+
+    [<Fact>]
+    let ``Obsolete attribute is taken into account when used on an type property and use extension method`` () =
+        Fsx """
+
+open System
+open System.Runtime.CompilerServices
+
+type Button = { [<Obsolete("Use B instead", true)>] Text : string }
+
+[<Extension>]
+type ButtonExtensions =
+
+    [<Extension>]
+    static member inline text(this: Button, text: string) =
+        { this with Text = text }
+        """
+        |> ignoreWarnings
+        |> compile
+        |> shouldFail
+        |> withErrorCode 101
+        |> withErrorMessage "This construct is deprecated. Use B instead"
+
+
+    [<Fact>]
+    let ``Obsolete attribute is taken into account when used on an type and property and use extension method`` () =
+        Fsx """
+
+open System
+open System.Runtime.CompilerServices
+
+[<Obsolete("Use B instead", true)>]
+type Button = { [<Obsolete("Use B instead", true)>] Text : string }
+
+[<Extension>]
+type ButtonExtensions =
+
+    [<Extension>]
+    static member inline text(this: Button, text: string) =
+        { this with Text = text }
+        """
+        |> ignoreWarnings
+        |> compile
+        |> shouldFail
+        |> withErrorCodes [ 101; 101]
```

Comment:
Please could you use withDiagnosticErrors like here:
https://github.com/dotnet/fsharp/blob/a65ace7698c159c34bd00f6408c7d4beb89b687d/tests/FSharp.Compiler.ComponentTests/Conformance/Expressions/BindingExpressions/BindingExpressions.fs#L24

````
        |> withDiagnostics [
            (Error 222, Line 6, Col 1, Line 7, Col 1, "Files in libraries or multiple-file applications must begin with a namespace or module declaration, e.g. 'namespace SomeNamespace.SubNamespace' or 'module SomeNamespace.SomeModule'. Only the last source file of an application may omit such a declaration.")
        ]
````

Reply:
Sure . I updated the rest of the test for consistency :)